src/ts/components/dashboard/TopMenu.tsx

```diff
@@ -38,7 +52,28 @@ class TopMenu extends React.Component<ITopMenuProps, ITopMenuState> {
     if (this.props.settings.selectedAccount) {
       this.props.getTransactions(this.props.settings.selectedAccount.address, data.value)
     }
+    const { apiContext } = this.props
+    this.setState({ isLoading: true })
+    apiContext.provider && this.props.destroyApi(apiContext.provider)
+  }
 
+  recreateApi = () => {
+    const { apiContext, settings } = this.props
+    if (!apiContext.apiReady && settings) {
```

Comment:
why do we need check `settings` here? If we don't have that, we can do 
```
if (apiContext.apiRead) {
} else {
}
```
which seems more readable